server/src/views/frame.js

```diff
@@ -115,14 +64,14 @@ class Snippet extends React.Component {
     }
 
     return <div className="snippet-container">
-      <a href={'#clip=' + encodeURIComponent(clip.id)}>
+      <a href={'?clip=' + encodeURIComponent(clip.id)}>
         { node }
       </a>
       <p>
         <img ref="commentBubble" className="comment-bubble"
           src={ closed ? this.props.staticLink("img/comment-bubble.png") : this.props.staticLink("img/comment-bubble-open.png") }
           onClick={ this.onClickComment.bind(this) } />
-        <a href={'#clip=' + encodeURIComponent(clip.id)}>
+        <a href={'?clip=' + encodeURIComponent(clip.id)}>
```

Comment:
We can't render the content highlighting on the server anyway, it is contained in an iframe.

You can cache the HTML, it just isn't a completely accurate representation of the clip number indicator.  The server HTML doesn't even contain the scroll position, nor can the server calculate where the highlight should appear.  So the only thing the server can add is a little indicator.  The active clip is primarily a client-side mechanism.